### DISCUSSION TITLE
running doctests

### DISCUSSION THREAD

**Discussion by jeromedockes** [skrub-data/skrub#780] [General]
shall we consider having a CI job running the doctests -- the small examples in the docstrings? otherwise they are likely to get outdated. for example the table vectorizer one [shows](https://github.com/skrub-data/skrub/blob/79d6bb397c93fa7666fa92711b246862bb332b31/skrub/_table_vectorizer.py#L491) a "remainder" key that now would be "numeric". testing the docstrings would also have caught #746

**Comment by Vincent-Maladiere** [comment]
Very good point. What would be your preferred way to do it?

**Comment by jeromedockes** [comment]
one easy way is to add the `--doctest-modules` to the pytest config as is done in #788 ; if that sounds reasonable I'll start addressing the failures there. If you have another preference LMK!

If at some point we start adding doctest snippets in the narrative documentation eg the user guide, we can also test those by setting the `--doctest-glob` option

**Comment by jeromedockes** [comment]
has been fixed
